toolkits/landcover/impl/dataset.js

```diff
@@ -183,6 +183,42 @@ Dataset.prototype.addBandIndices = function(var_names) {
 };
 
 
+/**
+ * Add the Tasseled Cap transformation to each image in the dataset.
+ *
+ * Each dataset will have it's own set of Tasseled Cap coefficients.
+ * The output bands will be named: [TC1, TC2, TC3, TC4, TC5, TC6];
+ *
+ * @return {!Dataset}
+ */
+Dataset.prototype.addTasseledCap = function() {
```

Comment:
Please add unit tests

Reply:
Unit test to test failure case.
Int test (in landsat8) to test the success case.  Should that be here instead (means we have to bring L8 into here, which doesn't seem right).

Reply:
Since this base class will be used by various implementations, we can test the base impl once by mocking out the needed functionality in tests - e.g., you could just return a set of fake values from `getTasseledCapCoefficients()` in the mocked instance.